packages/react-instantsearch-core/src/connectors/connectHits.js

```diff
@@ -45,9 +46,19 @@ export default createConnector({
 
   getProvidedProps(props, searchState, searchResults) {
     const results = getResults(searchResults, this.context);
-    const hits = results ? results.hits : [];
-
-    return { hits };
+    if (!results) {
+      return { hits: [] };
+    }
+    const hitsWithPositions = addAbsolutePositions(
+      results.hits,
+      results.hitsPerPage,
+      results.page
+    );
+    const hitsWithPositionsAndQueryID = addQueryID(
```

Comment:
these names don't seem very useful to me, what about composing it: 

```js
  return {
    hits: addQueryId(
      addAbsolutePositions(
        results.hits,
        results.hitsPerPage,
        results.page
      ),
      results.queryID
    )
  };
```

Reply:
@samouss advised to keep the names more explicit in his review. They do look at little verbose. I have no preference.